packages/request-client.js/src/api/request.ts

```diff
@@ -674,7 +681,9 @@ export default class Request {
         )}`,
       );
     }
-    if (this.paymentNetwork && requestAndMeta.result.request) {
+    if (this.skipPaymentDetection) {
+      this.balance = null;
+    } else if (this.paymentNetwork && requestAndMeta.result.request) {
```

Comment:
what if `skipPaymentDetection === false` and `requestAndMeta.result.request === null`. Will `balance` be undefined? is this wished?

Reply:
after rethinking it, we will return the previous balance (to avoid loosing data)